src/main/java/com/mercateo/spring/security/jwt/security/JWTAuthenticationTokenFilter.java

```diff
@@ -15,40 +15,64 @@
  */
 package com.mercateo.spring.security.jwt.security;
 
+import com.mercateo.spring.security.jwt.token.exception.InvalidTokenException;
+
 import java.io.IOException;
 
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.security.authentication.AnonymousAuthenticationProvider;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
 import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
 import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.util.AntPathMatcher;
 
-import com.mercateo.spring.security.jwt.token.exception.InvalidTokenException;
-
-import lombok.extern.slf4j.Slf4j;
+import io.vavr.collection.Set;
 
 @Slf4j
 public class JWTAuthenticationTokenFilter extends AbstractAuthenticationProcessingFilter {
 
     private final static String TOKEN_HEADER = "authorization";
 
-    public JWTAuthenticationTokenFilter() {
+    private final AntPathMatcher antPathMatcher =  new AntPathMatcher();
+
+    @NonNull
+    private Set<String> unauthenticatedPaths;
+
+    public JWTAuthenticationTokenFilter(@NonNull Set<String> unauthenticatedPaths) {
         super("/**");
+        this.unauthenticatedPaths = unauthenticatedPaths;
     }
 
     @Override
     public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) {
-        String header = request.getHeader(TOKEN_HEADER);
+        String tokenHeader = request.getHeader(TOKEN_HEADER);
+
+        if (tokenHeader == null || !tokenHeader.startsWith("Bearer ")) {
+            final String pathInfo = String.valueOf(request.getPathInfo()).replace("null","");
```

Comment:
I'd prefer a dedicated method for this large block (`getAuthenticationWithoutToken(request, tokenHeader)`).